### PR TITLE
artalk: init at 2.8.6

### PR DESCRIPTION
**Pull request by Moraxyc**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Moraxyc** [comment]
Result of `nixpkgs-review pr 308423` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>artalk</li>
  </ul>
</details>

File: pkgs/by-name/ar/artalk/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  ];
+
+  passthru.tests = {
+    version = testers.testVersion {
+      package = artalk;
+      command = "${artalk}/bin/Artalk --version";
+    };
+  };
+
+  meta = with lib; {
```
Comment (TomaSajt) [inline pkgs/by-name/ar/artalk/package.nix:37]: Please set `meta.mainProgram` to `"Artalk"`.
Among other things, this makes `testers.testVersion` run the correct command by itself, so you can remove the explicit command call.

Comment (Moraxyc) [reply pkgs/by-name/ar/artalk/package.nix:34]: oh I forgot it
:(
Thx

**Comment by Moraxyc** [comment]
Result of `nixpkgs-review pr 308423` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>artalk</li>
  </ul>
</details>

**Comment by Moraxyc** [comment]
@ofborg test artalk

**Comment by Moraxyc** [comment]
@ofborg test artalk

**Comment by Moraxyc** [comment]
@ofborg test artalk

**Comment by Moraxyc** [comment]
Due to the frontend of Artalk is built by pnpm which is not reproducible enough for nix right now, I’d like to request a separate frontend release from upstream.

**Comment by Moraxyc** [comment]
@ofborg test artalk

File: nixos/modules/services/web-apps/artalk.nix [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            ''
+              cat /run/artalk/new | ${pkgs.yj}/bin/yj -r > "${cfg.configFile}"
+              rm /run/artalk/new
+            ''
+        );
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        Type = "simple";
+        ExecStart = "${cfg.package}/bin/Artalk server --config ${cfg.configFile} --workdir ${cfg.workdir} --host ${cfg.settings.host} --port ${builtins.toString cfg.settings.port}";
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/artalk.nix:120]:
```suggestion
        ExecStart = "${lib.getExe cfg.package} server --config ${cfg.configFile} --workdir ${cfg.workdir} --host ${cfg.settings.host} --port ${builtins.toString cfg.settings.port}";
```

File: nixos/modules/services/web-apps/artalk.nix [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            ''
+              [ -e "${cfg.configFile}" ] || ${cfg.package}/bin/Artalk gen config "${cfg.configFile}"
+              cat "${cfg.configFile}" | ${pkgs.yj}/bin/yj > "/run/artalk/old"
+              ${pkgs.jq}/bin/jq -s '.[0] * .[1]' "/run/artalk/old" "/run/artalk/new" > "/run/artalk/result"
+              cat "/run/artalk/result" | ${pkgs.yj}/bin/yj -r > "${cfg.configFile}"
+              rm /run/artalk/{old,new,result}
+            ''
+          else
+            ''
+              cat /run/artalk/new | ${pkgs.yj}/bin/yj -r > "${cfg.configFile}"
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/artalk.nix:112]:
```suggestion
              cat /run/artalk/new | ${lib.getExe pkgs.yj} -r > "${cfg.configFile}"
```

**Pull request merged**